assembly/pom.xml

```diff
@@ -136,6 +136,10 @@
           <artifactId>spark-yarn_${scala.binary.version}</artifactId>
           <version>${project.version}</version>
         </dependency>
+        <dependency>
+          <groupId>org.apache.hadoop</groupId>
```

Comment:
This artifact is only needed at runtime so included here in distribution. Ideally we should declare it as a runtime dependency. However, the `WebAppProxyServlet` class in the jar conflicts with the same one from `hadoop-cluster-client-minicluster` during tests.